### ISSUE TITLE
Can't call a generic function twice with the same data type

### ISSUE THREAD

**Issue by progamateur** [vlang/v#3074]
**V version:** V 0.1.23 02c2962
**OS:**  Windows 10 (1909)

**What did you do?**
I tried to compile the following:
 ```
fn main() {  
	println(simple_generic_fn('01'))    // call with string type
	println(simple_generic_fn( 23 ))   //	call with int type
	println(simple_generic_fn( 4.5))   // call with f32 type
	println(simple_generic_fn( 67 ))   // again, call with int type 
}

fn simple_generic_fn<T>(var T) T  {
	return var
}
```
**What did you expect to see?**
01
23
4.500000
67
 
**What did you see instead?**

e1.v:5:33: `T` needs to have method `str() string` to be printable
    3|  println(simple_generic_fn( 23 )) // call with int type
    4|  println(simple_generic_fn( 4.5)) // call with f32 type
    5|  println(simple_generic_fn( 67 )) // again call with int type 
                                         ^
    6| }
    7|

**Comment by medvednikov** [comment]
Very interesting find, thanks!

**Comment by Kamori** [comment]
Weird this works

```
fn main() {  
   println(simple_generic_fn('01'))    // call with string type
   println(simple_generic_fn( 23 ))   //	call with int type
   println(simple_generic_fn( 4.5))   // call with f32 type
}

fn simple_generic_fn<T>(var T) T  {
   return var
}
```

this doesn't

```
fn main() {  
   println(simple_generic_fn('01'))    // call with string type
   println(simple_generic_fn( 23 ))   //	call with int type
   println(simple_generic_fn( 4.5))   // call with f32 type
   println(simple_generic_fn( 'anything'))   // again, call with int type 
}

fn simple_generic_fn<T>(var T) T  {
   return var
}
```

**Comment by joe-conigliaro** [comment]
thanks @progamateur I'm on it

**Comment by joe-conigliaro** [comment]
@progamateur this is fixed in #3081 Thanks for reporting!
